Dalamud/Interface/Internal/Windows/Data/Widgets/NetworkMonitorWidget.cs

```diff
@@ -0,0 +1,172 @@
+    }
+
+    private void DrawNetworkPacket(NetworkPacketData data)
+    {
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted(data.Direction.ToString());
+
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted(this.opCodeDict.TryGetValue(data.OpCode, out var pair) ? pair.Item1 : data.OpCode.ToString());
+
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted($"0x{data.SourceActorId:X}");
+
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted($"0x{data.TargetActorId:X}");
+
+        ImGui.TableNextColumn();
+        if (data.Data.Count > 0)
+        {
+            ImGui.TextUnformatted(string.Join(" ", data.Data.Select(b => b.ToString("X2"))));
+        }
+    }
+
+    private void DrawFilterInput()
+    {
+        var invalidRegEx = this.filterString.Length > 0 && this.trackedOpCodes == null;
+        using var style = ImRaii.PushStyle(ImGuiStyleVar.FrameBorderSize, 2 * ImGuiHelpers.GlobalScale, invalidRegEx);
+        using var color = ImRaii.PushColor(ImGuiCol.Border, 0xFF0000FF, invalidRegEx);
+        ImGui.SetNextItemWidth(ImGui.GetContentRegionAvail().X);
+        if (!ImGui.InputTextWithHint("##Filter", "Regex Filter OpCodes...", ref this.filterString, 256))
+        {
+            return;
+        }
+
+        if (this.filterString.Length == 0)
+        {
+            this.trackedOpCodes = null;
+        }
+        else
+        {
+            try
+            {
+                this.trackedOpCodes = new Regex(this.filterString, RegexOptions.Compiled | RegexOptions.ExplicitCapture);
+            }
+            catch
+            {
+                this.trackedOpCodes = null;
+            }
+        }
+    }
+
+    private void OnNetworkMessage(nint dataPtr, ushort opCode, uint sourceActorId, uint targetActorId, NetworkMessageDirection direction)
+    {
+        if (this.trackedOpCodes == null || this.trackedOpCodes.IsMatch(this.OpCodeToString(opCode)))
+        {
+            this.packets.Enqueue(new NetworkPacketData(this, opCode, direction, sourceActorId, targetActorId, dataPtr));
+            while (this.packets.Count > this.trackedPackets)
+            {
+                this.packets.TryDequeue(out _);
+            }
+        }
+    }
+
+    private int GetSizeFromOpCode(ushort opCode)
+        => this.opCodeDict.TryGetValue(opCode, out var pair) ? pair.Item2 : 0;
+
+    /// <remarks> Add known packet-name -> packet struct size associations here to copy the byte data for such packets. </remarks>>
+    private int GetSizeFromName(string name)
+        => name switch
+        {
+            _ => 0,
+        };
+
+    /// <remarks> The filter should find opCodes by number (decimal and hex) and name, if existing. </remarks>
+    private string OpCodeToString(ushort opCode)
+        => this.opCodeDict.TryGetValue(opCode, out var pair) ? $"{pair.Item1}\0{opCode}\0{opCode:X}" : $"{opCode}\0{opCode:X}";
```

Comment:
Should probably change the order of `pair.Item1` and `opCode` to mage Negative Lookahead with numbers easier for filtering.